classes/Deck.js

```diff
@@ -1,22 +1,115 @@
-import { PlayingCard } from './PlayingCard'
+import { PlayingCard, SUITES, RANKS } from './playing-card'
 
 class Deck {
-  constructor(suites, ranks) {
+  constructor(cards) {
+    this.cards = cards
+  }
+
+  static from(suites, ranks) {
     const cards = []
+    let i = 0
     for (const suite of suites) {
       for (const rank of ranks) {
-        cards.push(new PlayingCard(suite, rank))
+        cards.push(new PlayingCard(suite, rank, i++))
       }
     }
-    this.cards = cards
+    return cards
+  }
```

Comment:
I don't understand the reason for this method. The name is misleading (I would think an instance of a Deck is returned from it), and it exposes a piece of its implementation (the use of PlayingCards) that is unnecessary to be known.

It makes more sense to me to have a setup with static methods that the user can choose between to create a Deck. For example
One where the user passes in 2 arrays of `suites` and `ranks` which you will then create full variations of the given values. Or another where the user passes in 1 array of objects with a `suite` and a `rank` field. Each object would designate a card that the user wishes to have in the Deck. Or any other setup that you think a user may want to pass in the relevant info.

But the point is is that you can now decide how you will construct the final array of cards (like constructing an array of `PlayingCard`s) and then call the Deck constructor with that array. Or you can just decide that the user must use one of these ways and use that as the constructor, like you had in the previous pull request.

Now the user never knows about a PlayingCard being used because they don't need to know that. They just need to know an acceptable way pass the relevant info to you.